src/openMVG/cameras/Camera_Pinhole.hpp

```diff
@@ -108,11 +108,13 @@ class Pinhole_Intrinsic : public IntrinsicBase
   // Data wrapper for non linear optimization (update from data)
   virtual bool updateFromParams(const std::vector<double> & params)
   {
-    if (params.size() == 3) {
-      *this = Pinhole_Intrinsic(_w, _h, params[0], params[1], params[2]);
+    if (params.size() == 3)
+    {
+      this->setK(params[0], params[1], params[2]);
```

Comment:
I don't see any replication, calling the constructor will be give more code and doesn't make "sense": we are not building a Camera, there is no reason to modify the width/height, etc.
And that was the problem: the constructor may do some other stuff than updating the parameters. In the popart branch, we have added a _serialNumber which was erased by the call of updateFromParams.